addOns/spider/src/main/java/org/zaproxy/addon/spider/SpiderParam.java

```diff
@@ -755,7 +756,9 @@ public void setHandleParameters(HandleParametersOption handleParametersVisited)
      *     HandleParametersOption}.
      */
     public void setHandleParameters(String handleParametersVisited) {
-        this.handleParametersVisited = HandleParametersOption.valueOf(handleParametersVisited);
+        HandleParametersOption option =
+                EnumUtils.getEnumIgnoreCase(HandleParametersOption.class, handleParametersVisited);
+        this.handleParametersVisited = option;
```

Comment:
Per:
https://github.com/zaproxy/zap-extensions/blob/726eee02818baf536732b78c80c5b7fa004ada6a/addOns/spider/src/main/java/org/zaproxy/addon/spider/SpiderParam.java#L230

this is initially set to USE_ALL, shall we default to USE_ALL when there's no match? (Along with a log message)

Reply:
It seems defaulting was done for [GraphQL](https://github.com/zaproxy/zap-extensions/pull/4726), so I'll do the same here.